### PR TITLE
Enable cache when building container images

### PR DESCRIPTION
**Pull request by d-hayashi**
## What?
Copy only `package.json` and `package-lock.json` but not the entire directory on the first stage of the Dockerfile

## Why?
This will enable build-cache and avoid re-installing dependencies except when you change `package.json` or `package-lock.json`

### CODE REVIEW

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-template-vite/pull/170#" title="Author self-approved">d-hayashi</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-template-vite).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-template-vite/blob/main/OWNERS)~~ [d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by yusukefs — APPROVED** [review]
LGTM!

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-template-vite/pull/170#" title="Author self-approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/app-template-vite/pull/170#pullrequestreview-893273372" title="Approved">yusukefs</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-template-vite).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-template-vite/blob/main/OWNERS)~~ [d-hayashi,yusukefs]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
